tracer/src/Datadog.Trace/AppSec/Waf/ReturnTypes.Managed/InitializationResult.cs

```diff
@@ -0,0 +1,92 @@
+// <copyright file="InitializationResult.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Datadog.Trace.AppSec.Waf.NativeBindings;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.AppSec.Waf.ReturnTypesManaged
+{
+    internal class InitializationResult
+    {
+        public InitializationResult(IntPtr? ruleHandle, ushort failedToLoadRules, ushort loadedRules, string ruleFileVersion, IReadOnlyDictionary<string, string[]> errors, bool unusableRuleFile = false)
+        {
+            HasErrors = errors.Count > 0;
+            Errors = errors;
+            ErrorMessage = HasErrors ? JsonConvert.SerializeObject(errors) : string.Empty;
+            FailedToLoadRules = failedToLoadRules;
+            LoadedRules = loadedRules;
+            RuleFileVersion = ruleFileVersion;
+            RuleHandle = ruleHandle;
+            UnusableRuleFile = unusableRuleFile;
+        }
+
+        internal IntPtr? RuleHandle { get; }
+
+        internal ushort FailedToLoadRules { get; }
+
+        /// <summary>
+        /// Gets the number of rules successfully loaded
+        /// </summary>
+        internal ushort LoadedRules { get; }
+
+        internal IReadOnlyDictionary<string, string[]> Errors { get; }
+
+        internal string ErrorMessage { get; }
+
+        internal bool HasErrors { get; }
+
+        internal bool UnusableRuleFile { get; }
+
+        internal string RuleFileVersion { get; }
+
+        internal string File { get; set; }
+
+        internal bool FileNotFound { get; set; }
+
+        internal bool ExceptionDecodingFile { get; set; }
```

Comment:
These 3 properties don't seem to be used anywhere. Can they be deleted?

Reply:
good catch fixed in 50c1359ab57b5cc2a9503358084816b10d950f65